tap_shopify/streams/base.py

```diff
@@ -190,3 +202,159 @@ def sync(self):
+            Returns the total number of records received and
+            emitted to target.
+        """
+        updated_at_min = self.get_bookmark()
+        end_date = Context.config.get("end_date", None)
+        updated_at_max = utils.strptime_with_tz(end_date) if end_date is not None else singer.utils.now().replace(microsecond=0)
+        results_per_page = int(Context.config.get("results_per_page", RESULTS_PER_PAGE))
+        
+        DT_FMT = '%Y-%m-%dT%H:%M:%S'
+        query_params = {
+            "updated_at_min": utils.strftime(updated_at_min, format_str=DT_FMT),
+            "updated_at_max": utils.strftime(updated_at_max, format_str=DT_FMT),
+            "status": "any"
+        }
+
+        recs_count = RunAsync.sync(
+            schema = self.schema,
+            stream_id = self.name,
+            endpoint = self.endpoint,
+            result_key = self.result_key,
+            params = query_params,
+            retry_limit = MAX_RETRIES,
+            results_per_page = results_per_page
+        )
+
+        self.update_bookmark(utils.strftime(updated_at_max))
+
+        return recs_count
+
+
+class RunAsync():
+    def __init__(self, schema, stream_id, endpoint, result_key, params, retry_limit, results_per_page):
+        self.schema = schema
+        self.stream_id = stream_id
+        self.endpoint = endpoint
+        self.result_key = result_key
+        self.params = params
+        self.retry_limit = retry_limit
+        self.results_per_page = results_per_page
+        
+        self.params['limit'] = str(self.results_per_page)
+        self.base_url = shopify.ShopifyResource.get_site()
+        self.current_shop = shopify.Shop.current()
+        self.bucket_size = 80 if self.current_shop.plan_name == "shopify_plus" else 40
+        self.shop_display_url = "https://{}".format(self.current_shop.myshopify_domain)
+        self.rec_count = 0
+        self.DT_FMT = '%Y-%m-%dT%H:%M:%S'
+
+    def get_hourly_chunks(self, start, end, num_hours=1):
+        ranges = []
+        NUM_SECONDS = num_hours * 60 * 60
+        st = utils.strptime_with_tz(start)
+        ed = utils.strptime_with_tz(end)
+        
+        while st < ed:
+            curr_ed = st + timedelta(seconds=NUM_SECONDS)
+            if curr_ed > ed:
+                curr_ed = ed
+            ranges.append({'updated_at_min': st, 'updated_at_max': curr_ed})
+            st = curr_ed + timedelta(seconds=1)
+
+        return ranges
+
+    async def _get_async(self, url, headers=None, params=None, retry_attempt=0):
+        headers = {**headers, "Connection": "close"} if headers else {"Connection": "close"}
+        try:
+            async with aiohttp.ClientSession() as session:
+                async with session.get(url=url, headers=headers, params=params) as response:
+                        if response.status == 200:
+                            return await response.json()
+                        elif response.status == 429:
```

Comment:
There appears to be a bug here with status codes != 200, 429, and 500-599. Getting a 401, for example will miss.